plugins/org.eclipse.elk.alg.layered/src/org/eclipse/elk/alg/layered/intermediate/PartitionPreprocessor.java

```diff
@@ -9,107 +9,84 @@
 package org.eclipse.elk.alg.layered.intermediate;
 
-import java.util.Collection;
-import java.util.Iterator;
-import java.util.List;
 import java.util.stream.Collectors;
 
 import org.eclipse.elk.alg.layered.graph.LEdge;
 import org.eclipse.elk.alg.layered.graph.LGraph;
-import org.eclipse.elk.alg.layered.graph.LNode;
-import org.eclipse.elk.alg.layered.graph.LPort;
-import org.eclipse.elk.alg.layered.options.InternalProperties;
 import org.eclipse.elk.alg.layered.options.LayeredOptions;
 import org.eclipse.elk.core.alg.ILayoutProcessor;
-import org.eclipse.elk.core.options.PortSide;
 import org.eclipse.elk.core.util.IElkProgressMonitor;
 
-import com.google.common.collect.HashMultimap;
-import com.google.common.collect.Multimap;
+import com.google.common.collect.Streams;
 
 /**
- * Add constraint edges between partitions. </br>
- * In partitioned graphs, nodes may have an assigned partition. For every such node holds, that every other node
- * with a smaller partition is placed left to it. This is accomplished by adding partition constraint
- * edges. For each node, we add an edge to every node in the next greater partition. These edges get a
- * high priority assigned such that during cycle breaking, none of these constraint edges gets reversed.
- * During layering, the constraint edges assure the aforementioned condition.
- * The constraint edges are removed later by the {@link PartitionPostprocessor}.
+ * Reverses edges that run contrary to layout partitions. More specifically, an edge (a, b) is reversed if a is
+ * configured for a higher layout partition than b. If all nodes have a partition set, the result is a graph that can
+ * only contain cycles among nodes in the same partition, and no edge reversed by this processor can be part of a cycle.
+ * Thus, the cycle breaker should not restore any such edge.
+ * 
+ * <p>
+ * If there are nodes that do not have a partition configured, that's another story. Since there can be arbitrarily
+ * many such nodes connected in arbitrary ways, any edge we reverse here could be part of a cycle and thus be restored
+ * again during cycle breaking. We try to avoid this by imposing a high direction priority on them, but we cannot
+ * guarantee success.
+ * </p>
  *
  * <dl>
  *   <dt>Precondition:</dt>
  *     <dd>an unlayered graph.</dd>
  *   <dt>Postcondition:</dt>
- *     <dd>unlayered graph with partition constraint edges.</dd>
+ *     <dd>edges that contradict layout partitions are reversed.</dd>
  *   <dt>Slots:</dt>
  *     <dd>Before phase 1.</dd>
  *   <dt>Same-slot dependencies:</dt>
  *     <dd>None.</dd>
  * </dl>
+ * 
+ * @see PartitionMidprocessor
+ * @see PartitionPostprocessor
  */
 public class PartitionPreprocessor implements ILayoutProcessor<LGraph> {
 
     /** The priority to set on added constraint edges. */
-    private static final int PARTITION_CONSTRAINT_EDGE_PRIORITY = 20;
+    private static final int PARTITION_CONSTRAINT_EDGE_PRIORITY = 50;
```

Comment:
Maybe add that it's an arbitrary value `>1`. The desire would be to set it to `INT_MAX`, however, since the value is summed up ... that would result in an overflow.